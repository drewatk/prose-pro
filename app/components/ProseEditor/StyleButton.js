import React from "react";
import styles from "./ProseEditor.css";

export default class StyleButton extends React.Component {
  constructor() {
    super();
    this.onToggle = e => {
      e.preventDefault();
      this.props.onToggle(this.props.style);
    };
  }

  render() {
    const { active, label } = this.props;
    return (
      <span
        className={`${styles.styleButton} ${active ? styles.activeButton : ""}`}
        onMouseDown={this.onToggle}
        role="button"
      >
        {label}
      </span>
    );
  }
}
