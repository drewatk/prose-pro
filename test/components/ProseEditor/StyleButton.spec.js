import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import StyleButton from "app/components/ProseEditor/StyleButton";

Enzyme.configure({ adapter: new Adapter() });

describe("StyleButton", () => {
  it("Matches Snapshot", () => {
    const wrapper = shallow(<StyleButton />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Matches Snapshot (active)", () => {
    const wrapper = shallow(<StyleButton active />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Toggles", () => {
    const onToggleMock = jest.fn();
    const wrapper = shallow(<StyleButton onToggle={onToggleMock} />);
    expect(wrapper.find(".styleButton")).toHaveLength(1);

    const mockEvent = { preventDefault: jest.fn() };
    wrapper.find(".styleButton").simulate("mousedown", mockEvent);

    expect(mockEvent.preventDefault).toHaveBeenCalledTimes(1);
    expect(onToggleMock).toHaveBeenCalledTimes(1);
  });
});
