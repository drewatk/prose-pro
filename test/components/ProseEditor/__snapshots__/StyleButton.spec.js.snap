// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleButton Matches Snapshot (active) 1`] = `
<span
  className="styleButton activeButton"
  onMouseDown={[Function]}
  role="button"
/>
`;

exports[`StyleButton Matches Snapshot 1`] = `
<span
  className="styleButton "
  onMouseDown={[Function]}
  role="button"
/>
`;
